docs/project/development-guide.md

```diff
@@ -1,72 +1,436 @@
+
+```sh
+make test-python-integration-local
+```
+
+#### (Advanced) Full integration tests
+To test across clouds, on top of setting up Redis, you also need GCP / AWS / Snowflake setup.
+
+> Note: you can manually control what tests are run today by inspecting
+> [RepoConfiguration](https://github.com/feast-dev/feast/blob/master/sdk/python/tests/integration/feature_repos/repo_configuration.py)
+> and commenting out tests that are added to `DEFAULT_FULL_REPO_CONFIGS`
+
+**GCP**
+1. You can get free credits [here](https://cloud.google.com/free/docs/free-cloud-features#free-trial).
+2. You will need to setup a service account, enable the BigQuery API, and create a staging location for a bucket.
+
+* Setup your service account and project using steps 1-5 [here](https://codelabs.developers.google.com/codelabs/cloud-bigquery-python#0).
+  * Remember to save your `PROJECT_ID` and your `key.json`. These will be your secrets that you will need to configure in Github actions. Namely, `secrets.GCP_PROJECT_ID` and `secrets.GCP_SA_KEY`. The `GCP_SA_KEY` value is the contents of your `key.json` file.
+* Follow these [instructions](https://cloud.google.com/storage/docs/creating-buckets) in your project to create a bucket for running GCP tests and remember to save the bucket name.
+  * Make sure to add the service account email that you created in the previous step to the users that can access your bucket. Then, make sure to give the account the correct access roles, namely `objectCreator`, `objectViewer`, `objectAdmin`, and `admin`, so that your tests can use the bucket.
+
+3. Install the [Cloud SDK](https://cloud.google.com/sdk/docs/install).
+4. Login to gcloud if you haven't already:
+  ```
+  gcloud auth login
+  gcloud auth application-default login
+  ```
+- When you run `gcloud auth application-default login`, you should see some output of the form:
+  ```
+  Credentials saved to file: [$HOME/.config/gcloud/application_default_credentials.json]
+  ```
+- You should run `export GOOGLE_APPLICATION_CREDENTIALS="$HOME/.config/gcloud/application_default_credentials.json”` to add the application credentials to your .zshrc or .bashrc.
+5. Run `export GCLOUD_PROJECT=[your project id from step 2]` to your .zshrc or .bashrc.
+6. Running `gcloud config list` should give you something like this:
+  ```sh
+  $ gcloud config list
+  [core]
+  account = [your email]
+  disable_usage_reporting = True
+  project = [your project id]
+
+  Your active configuration is: [default]
+  ```
+7. Export GCP specific environment variables in your workflow. Namely,
+  ```sh
+  export GCS_REGION='[your gcs region e.g US]'
+  export GCS_STAGING_LOCATION='[your gcs staging location]'
+  ```
+**NOTE**: Your `GCS_STAGING_LOCATION` should be in the form `gs://<bucket name>` where the bucket name is from step 2.
+
+8. Once authenticated, you should be able to run the integration tests for BigQuery without any failures.
 
-## Feast Protobuf API
```

Comment:
this seems to have been completed deleted; was that intentional?

Reply:
Yea I souped up the protobuf section elsewhere in this doc but this snipped wasn't super useful.